apps/files/src/components/FileSharingSidebar.vue

```diff
@@ -259,6 +280,22 @@ export default {
     noResharePermsMessage () {
       const translated = this.$gettext('You don\'t have permission to share this %{type}.')
       return this.$gettextInterpolate(translated, { type: this.highlightedFile.type }, false)
+    },
+
+    currentUsersPermissions () {
+      if (this.incomingShares.length > 0) {
+        let permissions = 1
+
+        for (const share of this.incomingShares) {
+          if (share.permissions > permissions) {
```

Comment:
`permissions` is a bitmask, isn't it? In that case we should not check if permission-int is higher, but instead just merge the permissions with `permissions |= share.permissions` to get the combination of all permissions of the user.

Reply:
Right - thanks! Fixed.